### ISSUE TITLE
SASS compiler issues on Node 8

### ISSUE THREAD

**Issue by CraigWarford** [trion-development/docker-ng-cli#5]
I've been using ng-cli-karma to run builds on bitbucket pipelines, but when things switched to node:8, the builds began to fail with a node-sass error:
`29 09 2017 00:30:38.713:ERROR [karma]: Error: Module build failed: Error: Missing binding /opt/atlassian/pipelines/agent/build/node_modules/node-sass/vendor/linux-x64-57/binding.node
Node Sass could not find a binding for your current environment: Linux 64-bit with Node.js 8.x
Found bindings for the following environments:
  - Linux 64-bit with Node.js 6.x
This usually happens because your environment has changed since running 'npm install'.
Run 'npm rebuild node-sass --force' to build the binding for your current environment.`

Locally, I installed NodeJS 8, and reproduced a similar error.  I ran the recommended command for `npm rebuild node-sass --force`, which rebuilt the items in the node_modules folder, and everything once again worked for ng building and testing.

On bitbucket, I actually stash node_modules.tar.gz in the Downloads, and use curl to pull that file and decompress it to speed the node_modules setup process.  So, I updated that file with the node-sass updates, and tried the build again.  It got a little further, but still failed on my ng test step:
`Error: Module build failed: Error: /usr/lib/x86_64-linux-gnu/libstdc++.so.6: version `GLIBCXX_3.4.21' not found (required by /opt/atlassian/pipelines/agent/build/node_modules/node-sass/vendor/linux-x64-57/binding.node)`

It appears that the node-sass is depending on GLIBCXX_3.4.21, but that's not accessible in the container.  Any ideas on how to make that file available?  Otherwise, I may have to fork this dockerfile and create a version that still uses node:6...

Thanks for taking a look.

**Comment by everflux** [comment]
Can you try to remove the node_modules folder before running `npm install` as part of your build?

**Comment by CraigWarford** [comment]
That did work, thanks so much.  I also created a to package the node_modules folder from that container, and now I'm able to use that.  Thanks again!

**Comment by everflux** [comment]
Glad that it worked out for you!
In generel I recommend to have a clean build to avoid problems like this. (I experienced problems when I had old versions of libraries in the node_modules folder as well.) If this slows down your build too much and you are not building in the cloud you can use a local npm mirror, yarn or the new caching facilities of npm 5. I would not use a container for to achieve package caching or stash the result and re-download it. Though I did not test npm 5 caching it should work like yarn which is described in the readme of this project.

**Comment by CraigWarford** [comment]
Thanks for the great suggestions.  We are indeed building in the cloud, on bitbucket.  I'm using a package lock file now, but I still haven't gotten to the point that I fully trust it yet.  I've been bitten pretty hard by sudden package incompatibilities (and actually, complete disappearances) in the past, which led me to keep a zip of the node_modules handy just in case.  When I realized that it was faster to curl that file and unzip it instead of doing the full npm install, I started doing it that way to save build minutes in the cloud.  But I'll continue with the fresh npm install and lock file, and keep the zip as a backup plan. (It was easy enough to recreate after getting it working in the container.)  Maybe the package lock file will prove sufficient protection. Time will tell.
